docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc

```diff
@@ -243,3 +243,78 @@ The removal of openshift-integration allows us to remove few thirdparty dependen
 any of these libraries as dependencies of your own providers deployed to Keycloak server, you may also need to copy those `jar` files
 explicitly to the Keycloak distribution `providers` directory as well.
 
+= Context and dependency injection no longer enabled to JAX-RS Resources
+
+In order to provide a better runtime and leverage as much as possible the underlying stack,
+all injection points for contextual data using the `javax.ws.rs.core.Context` annotation were removed. The expected improvement
+in performance involves no longer creating proxies instances multiple times during the request lifecycle, and drastically reducing the amount of reflection code at runtime.
+
+If you are extending one of the following SPIs:
+
+* `PolicySpi`
+* `AdminRealmResourceSpi`
+* `IdentityProviderSpi`
+* `RealmResourceProvider`
```

Comment:
```suggestion
* `RealmResourceSPI`
```
Should be like this when we're talking about SPIs, right?

Reply:
Done